src/main/java/com/google/sps/servlets/ChallengesServlet.java

```diff
@@ -92,16 +96,71 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(json);
   }
 
+  /* remove challenge id from user status map
+     add new challenge id to user status map
+     add challenge id to user completed challenge map */
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Is this replacing the functionality related to updating statuses in the user servlet? Or only when a completely new challenge is requested? If it overlaps with the user serlvet probably want to remove the part it is replacing

Reply:
oh no this is only when a new challenge is requested.

Reply:
i'll clarify my comment in the next commit

Reply:
So the doPut main purpose is to update User's 'completed challenges' and find and add a new challenge to 'challenge statuses'.